include/qpdf/QUtil.hh

```diff
@@ -535,6 +536,12 @@ namespace QUtil
     // purposes.
     QPDF_DLL
     size_t get_max_memory_usage();
+
+    // Temporary methods to be removed in qpdf 12.
```

Comment:
Why are they being removed in qpdf 12? What changes do you anticipate in qpdf 12 that will make these no longer needed?

Reply:
There are existing versions of these methods that take std::string const& parameters which can be transparently changed to take string_view parameters (which cover string_view, std::strings and C strings) except this cannot happen until  qpdf 12 as it is ABI breaking.

Reply:
It is not possible to overload the existing method with versions that take string_views as the compiler is unable to work out which version to use.

Reply:
The way I would do this (and request for you to do it) is, for each function, change to something like this:
```c++
    // ABI: change std::string const& to std::string_view (API-compatible) and remove the temporary view version.
    QPDF_DLL
    bool utf8_to_pdf_doc(
        std::string const& utf8, std::string& pdfdoc, char unknown_char = '?');
    // temporary until next ABI change
    bool utf8_view_to_pdf_doc(
        std::string_view utf8, std::string& pdfdoc, char unknown_char = '?');
```
See `ABI Changes` in TODO. Following this convention makes it very easy to find and address these the next time the ABI window opens up.